From 9a9a3bdd22bde7c75303c6bd980b63fba18318bf Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Tue, 17 Oct 2023 20:05:52 +0200
Subject: [PATCH 400/841] fix(psci): fix parent_idx in
 psci_validate_state_coordination

Update parent_idx support in psci_validate_state_coordination() as
it is done in psci_do_state_coordination(). The modified loop verifies
the targeted state for all the branch up to end_pwrlvl in the topology
for the current cpu.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 66d779ddc939d8c72a6a3ffa28c6df3bd0922c3c fix(psci): fix parent_idx in psci_validate_state_coordination]
Fixes: 110569d745db ("feat(psci): add support for OS-initiated mode")
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I14420f64a18b543eb4e10a1279f51cc17558c13c
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/334406
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 lib/psci/psci_common.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/psci/psci_common.c b/lib/psci/psci_common.c
index f9de432b2..c6f6a37a4 100644
--- a/lib/psci/psci_common.c
+++ b/lib/psci/psci_common.c
@@ -662,6 +662,8 @@ int psci_validate_state_coordination(unsigned int end_pwrlvl,
 			}
 			goto exit;
 		}
+
+		parent_idx = psci_non_cpu_pd_nodes[parent_idx].parent_node;
 	}
 
 	/*
-- 
2.47.2

